### ISSUE TITLE
missings tabs for an example code in docs

### ISSUE THREAD

**Issue by anasfik** [reactjs/reactjs.org#4862]
in Render and Commit page
on step 3 for clock example, there is tabs missing to show, 
https://beta.reactjs.org/learn/render-and-commit

**Comment by Rec0iL99** [comment]
@gaearon @harish-sethuraman could I work on this?

**Comment by Rec0iL99** [comment]
@anasfik there's a reason why the `index.js` tab is not shown in `Step 3`. 

It's because step 3's example is not similar to the other step's examples. Here the example uses state updates to explain re-renders. The logic implementation of the `Clock` component is hidden in a `App` which is only visible in the docs source code and not in the UI. 

```js
// App.js
import {useState, useEffect} from 'react';
import Clock from './Clock.js';

function useTime() {
  const [time, setTime] = useState(() => new Date());
  useEffect(() => {
    const id = setInterval(() => {
      setTime(new Date());
    }, 1000);
    return () => clearInterval(id);
  }, []);
  return time;
}

export default function App() {
  const time = useTime();
  return <Clock time={time.toLocaleTimeString()} />;
}
```

This is so as to avoid confusing/making the docs too overwhelming for someone who is completely **new** to React. Hence, in the UI only the `Clock.js` tab is included.

If we were to include the `index.js` tab then it would look like this:
```js
// index.js
import App from './App.js';
import ReactDOM from 'react-dom';

ReactDOM.render(<App />, document.getElementById('root'));
```
the person reading the doc would have no idea, as to what `App` component does and we would have to show the `App.js` tab where `useState` and `useEffect` hooks are being used and the person reading has no idea what these hooks do thus making the docs hard to follow.

I hope this clears things out.

@eps1lon this is my analysis of the issue and I feel it can be closed since it requires no fixing.

**Comment by harish-sethuraman** [comment]
Thanks @Rec0iL99

**Comment by anasfik** [comment]
well cleared, thanks
